app/src/main/java/com/sdp13epfl2021/projmag/activities/ProjectCreationActivity.kt

```diff
@@ -37,22 +37,24 @@ class Form : AppCompatActivity() {
         private const val REQUEST_SELECTION_ACCESS = 4
     }
 
+    private var projectToEdit: ImmutableProject? = null
+    private var changedVid = false
 
     //tag selection related variables
     private lateinit var tagRecyclerView: RecyclerView
 
-
     //video related variables
-
     private var videoUri: Uri? = null
     private var subtitles: String? = null
-
     private var listTags: Array<String> = emptyArray()
     private var listSections: Array<String> = emptyArray()
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_project_creation)
+
+        projectToEdit = intent.extras?.getParcelable("edit") as ImmutableProject?
```

Comment:
projectToEdit = intent.getParcelableExtra("edit") is shorter,
and we should try to avoid literal string for the intent name ("edit").